### PR TITLE
INSTUI-2876 - refactor themeOverrides to be handled in withStyle

### PR DESCRIPTION
**Pull request by ke1k0**
Closes: INSTUI-2876

The handling of `themeOverride` props were moved to the `withStyle` decorator. This way the theme.js
of the components are simplified.

### CODE REVIEW

**Review by HerrTopi — APPROVED** [review]
This was a nice catch. It simplifies theming even further, good job :)

File: packages/ui-editable/src/InPlaceEdit/styles.js [hunk 1]
```diff
@@ -48,7 +48,6 @@ const generateStyle = (componentTheme, props, state) => {
       touchAction: 'manipulation',
       background: 'transparent',
       border: 'none',
-      fontFamily: componentTheme.fontFamily,
```
Comment (matyasf) [inline packages/ui-editable/src/InPlaceEdit/styles.js:51]: Why was this deleted?

Comment (ke1k0) [reply packages/ui-editable/src/InPlaceEdit/styles.js:51]: Because there was no corresponding themeVariable set for it in the theme.js. I looked up the PR from before, remembered we had a discussion about it: https://github.com/instructure/instructure-ui/pull/328. The fontFamily was deleted from the theme.js then, so this can be deleted too.

Comment (matyasf) [reply packages/ui-editable/src/InPlaceEdit/styles.js:51]: I see, thanks!

**Review by matyasf — APPROVED** [review]
